router/http_test.go

```diff
@@ -339,6 +339,34 @@ func (s *S) TestHTTPHeadersFromClient(c *C) {
 	c.Assert(res.StatusCode, Equals, 200)
 }
 
+func (s *S) TestConnectionCloseHeaderFromClient(c *C) {
+	h := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// Connection: close header should be stripped by the reverse proxy so it
+		// always does keep-alive with backends.
+		c.Assert(req.Close, Equals, false)
+		c.Assert(req.Header.Get("Proxy-Authenticate"), Equals, "")
+		c.Assert(req.Header.Get("Proxy-Authorization"), Equals, "")
```

Comment:
And in the "Changes from RFC2616" section:

> Also, "hop-by-hop" header fields are required to appear in the Connection header field; just because they're defined as hop-by-hop in this specification doesn't exempt them.